guard/resources/validate/functions/rules/complex_rules.guard

```diff
@@ -0,0 +1,22 @@
+rule compare_number_of_buckets(expected) {
+    %expected == 5
+}
+
+let buckets = Resources.*[ Type == 'AWS::S3::Bucket' ]
+
+rule COMBINED_FUNCTION_AND_PARAMETERIZED_RULES when %buckets !empty {
+    let number_of_buckets = count(%buckets)
+    compare_number_of_buckets(%number_of_buckets)
```

Comment:
This is difficult to read, can we replace it with simple comparison with constant output? 

like:
```
count(%buckets) == 5
```